### PR TITLE
Added structured logging

### PR DESCRIPTION
**Pull request by Expecho**
Implemented structured logging (see #5865) for the project OrleansProviders. Doing it project by project makes it a bit more manageable than to do the whole lot in one changeset.

### CODE REVIEW

File: src/OrleansProviders/Storage/MemoryStorage.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ public MemoryGrainStorage(string name, MemoryGrainStorageOptions options, ILogge
             this.grainFactory = grainFactory;
 
             //Init
-            logger.Info("Init: Name={0} NumStorageGrains={1}", name, this.options.NumStorageGrains);
+            logger.Info("Init: Name={Name} NumStorageGrains={NumStorageGrains}", name, this.options.NumStorageGrains);
```
Comment (ReubenBond) [inline src/OrleansProviders/Storage/MemoryStorage.cs]: Use `LogInformation` instead of `Info` here and similarly elsewhere

Comment (Expecho) [reply src/OrleansProviders/Storage/MemoryStorage.cs]: > Use `LogInformation` instead of `Info` here and similarly elsewhere

Done!

**Comment by ReubenBond** [comment]
Thank you, Peter, much appreciated!

**Comment by Expecho** [comment]
> Thank you, Peter, much appreciated!

You're welcome. Now I dipped  my toes in it (and noticed the additional changes to look for) what would be the next best project in the solution to tackle?

**Comment by JanEggers** [comment]
@ReubenBond , @Expecho  did you consider this to squeeze out some more performance?

https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?view=aspnetcore-3.0

**Comment by Expecho** [comment]
> @ReubenBond , @Expecho did you consider this to squeeze out some more performance?
> 
> https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?view=aspnetcore-3.0

I get the point, but it might requires some upfront design regarding the eventids to make sure there is some consistency.

Work could be split in two seperate issues. First move to structured logging and after that see where we could benefit from performance.

**Comment by srollinet** [comment]
@JanEggers interesting link!

I only wonder it it will make this kind of pattern less maintainable
```
if (logger.IsEnabled(LogLevel.Trace)) logger.LogTrace("Read Keys={Keys}", StorageProviderUtils.PrintKeys(keys));
```
Because using `LoggerMessage.Define`, the LogLevel is defined statically, so you must check at 2 different places to ensure the LogLevels are consistent.

But this is not a big issue

**Comment by JanEggers** [comment]
@srollinet you actually can remove the if in usercode because its in the box:

https://github.com/aspnet/Extensions/blob/66007df6e5b68683de0cab0dc6a8552265318f20/src/Logging/Logging.Abstractions/src/LoggerMessage.cs#L100-L116

**Comment by JanEggers** [comment]
on the other hand StorageProviderUtils.PrintKeys should not be used in a structured logging approach as it removes the structure of the argument so it would be better to just pass keys without formatting them.

**Comment by srollinet** [comment]
@JanEggers yes, looking the Orleans code, this pattern is used a lot and it's probably to avoid parameters evaluation.

And yes, with structured logging there are maybe other ways to format the parameters :)

**Pull request merged**